src/rgen/Microsoft.Macios.Bindings.Analyzer/NativeObjectHandleAnalyzer.cs

```diff
@@ -0,0 +1,172 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Microsoft.Macios.Bindings.Analyzer;
+
+[DiagnosticAnalyzer (LanguageNames.CSharp)]
```

Comment:
I think it would be nice to document this analyzer, either in a comment here or a separate README.md:

* Explain which problem it's trying to solve, and how it tries to solve it.
* Current known false positives / limitations:
	* Doesn't understand arrays 
	* Only looks in the current scope for usage of a variable
	* `using (var ...)` vs `using var ...`
	* Anything else?

Reply:
Agreed.

`using (var ...)` vs `using var ...` is already solved, the rest is correct.

I think there are two more limitations:
- We don't skip over `Handle = ..` setters. This doesn't trigger on `this.Handle = ...` because `this` is considered to be implicitly kept alive. It also doesn't affect `InitializeHandle` calls. I think there was only one false positive from this so it was not worth the extra logic.
- We currently don't analyze code in the `Wrap` attribute (unlikely to be ever implemented as part of the analyzer).

Reply:
We shouldn't call the `Handle` setter anyway (we even have a test for this: https://github.com/dotnet/macios/blob/main/tests/cecil-tests/SetHandleTest.cs, but I haven't fixed all the existing cases yet), so this limitation doesn't need to be solved.